src/docker-in-docker/install.sh

```diff
@@ -175,72 +177,72 @@ fi
-    if [ "${MOBY_BUILDX_VERSION}" = "latest" ]; then
+    if [ "${package_version}" = "latest" ] || [ "${package_version}" = "lts" ] || [ "${package_version}" = "stable" ]; then
         # Empty, meaning grab whatever "latest" is in apt repo
-        buildx_version_suffix=""
+        echo ""
     else
-        buildx_version_dot_escaped="${MOBY_BUILDX_VERSION//./\\.}"
-        buildx_version_dot_plus_escaped="${buildx_version_dot_escaped//+/\\+}"
-        buildx_version_regex="^(.+:)?${buildx_version_dot_plus_escaped}([\\.\\+ ~:-]|$)"
-        set +e
-            buildx_version_suffix="=$(apt-cache madison moby-buildx | awk -F"|" '{print $2}' | sed -e 's/^[ \t]*//' | grep -E -m 1 "${buildx_version_regex}")"
+        # Fetch a valid version from the apt-cache (eg: the Microsoft repo appends +azure, breakfix, etc...)
+        # Regex needs to handle debian package version number format: https://www.systutorials.com/docs/linux/man/5-deb-version/
+        local package_version_dot_escaped="${package_version//./\\.}"
+        local package_version_dot_plus_escaped="${package_version_dot_escaped//+/\\+}"
+        local package_version_regex="^(.+:)?${package_version_dot_plus_escaped}([\\.\\+ ~:-]|$)"
+
+        set +e # Catch errors so we can show more useful output
+        local package_version_suffix="=$(apt-cache madison ${package_name} | awk -F"|" '{print $2}' | sed -e 's/^[ \t]*//' | grep -E -m 1 "${package_version_regex}")"
         set -e
-        if [ -z "${buildx_version_suffix}" ] || [ "${buildx_version_suffix}" = "=" ]; then
-            err "No full or partial moby-buildx version match found for \"${MOBY_BUILDX_VERSION}\" on OS ${ID} ${VERSION_CODENAME} (${architecture}). Available versions:"
-            apt-cache madison moby-buildx | awk -F"|" '{print $2}' | grep -oP '^(.+:)?\K.+'
+
+        if [ -z "${package_version_suffix}" ] || [ "${package_version_suffix}" = "=" ]; then
+            err "No full or partial ${package_name} version match found for \"${package_version}\" on OS ${ID} ${VERSION_CODENAME} (${architecture}). Available versions:"
+            apt-cache madison ${package_name} | awk -F"|" '{print $2}' | grep -oP '^(.+:)?\K.+'
             exit 1
         fi
-        echo "buildx_version_suffix ${buildx_version_suffix}"
+
+        echo "${package_name} package version suffix: ${package_version_suffix}"
+        echo $package_version_suffix
     fi
-fi
+}
 
 # Install Docker / Moby CLI if not already installed
 if type docker > /dev/null 2>&1 && type dockerd > /dev/null 2>&1; then
     echo "Docker / Moby CLI and Engine already installed."
 else
+    # Determine required packages and versions
+    engine_version_suffix=get_package_version_suffix "${engine_package_name}" "${DOCKER_VERSION}"
+    cli_version_suffix=get_package_version_suffix "${cli_package_name}" "${DOCKER_VERSION}"
+    required_packages="${cli_package_name}${cli_version_suffix} ${engine_package_name}${engine_version_suffix}"
+
+    # Moby always uses buildx
```

Comment:
Is this true?

Reply:
I don't think so! 

Reasoning :`buildx` is a Docker CLI plugin that extends the docker build command with the full support of the features provided by Moby BuildKit builder toolkit. However, `buildx` is not the default build engine for Docker or Moby. The default builder in Docker is still the legacy `docker build` command. To use `buildx`, you need to either install it as a Docker CLI plugin or use a version of Docker that has buildx bundled with it (Docker 19.03 and later versions).

So, while Moby and Docker can use `buildx` for building images, it's not always used and depends on the specific configuration and version of Docker you're using.

(Feel free to correct if I have misunderstood)